### PR TITLE
allow anonymous uuid generation to be customizable

### PR DESCRIPTION
**Pull request by jimexist**
## :bulb: Motivation and Context

this makes it aligned with the js/ts sdk

- https://github.com/PostHog/posthog-js/blob/55694356e168568700f3370f26d636cbe09bf795/src/posthog-core.ts#L169
- https://github.com/PostHog/posthog-ios/pull/133

## :green_heart: How did you test it?

unit test

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] I updated the docs if needed.
- [ ] No breaking change or entry added to the changelog.

### CODE REVIEW

File: posthog/src/main/java/com/posthog/PostHogConfig.kt [hunk 1]
```diff
@@ -94,6 +95,11 @@ public open class PostHogConfig(
      * The hook is called before the event is cached or sent over the wire
      */
     public var propertiesSanitizer: PostHogPropertiesSanitizer? = null,
+    /**
+     * Hook that allows for modification of the default mechanism for
+     * generating anonymous device id (which as of now is just random UUID v4)
+     */
+    public var getDeviceId: ((UUID) -> UUID) = { it },
```
Comment (marandaneto) [inline posthog/src/main/java/com/posthog/PostHogConfig.kt:102]: Same as https://github.com/PostHog/posthog-ios/pull/133/files#r1606325584

Comment (jimexist) [reply posthog/src/main/java/com/posthog/PostHogConfig.kt]: fixed an pushed

File: posthog/src/test/java/com/posthog/PostHogTest.kt [hunk 2]
```diff
@@ -852,6 +854,19 @@ internal class PostHogTest {
         sut.close()
     }
 
+    @Test
+    fun `allows for modification of the uuid generation mechanism`() {
+        val expected = UUID.randomUUID()
+        val config =
+            PostHogConfig(API_KEY, getDeviceId = {
```
Comment (marandaneto) [inline posthog/src/test/java/com/posthog/PostHogTest.kt:861]:
```suggestion
            PostHogConfig(API_KEY, getAnonymousId = {
```

File: posthog/src/main/java/com/posthog/PostHogConfig.kt [hunk 3]
```diff
@@ -94,6 +95,11 @@ public open class PostHogConfig(
      * The hook is called before the event is cached or sent over the wire
      */
     public var propertiesSanitizer: PostHogPropertiesSanitizer? = null,
+    /**
+     * Hook that allows for modification of the default mechanism for
+     * generating anonymous device id (which as of now is just random UUID v4)
+     */
```
Comment (marandaneto) [inline posthog/src/main/java/com/posthog/PostHogConfig.kt:101]:
```suggestion
    /**
     * Hook that allows for modification of the default mechanism for
     * generating anonymous id (which as of now is just random UUID v4)
     */
```

**Comment by marandaneto** [comment]
@jimexist thanks for the PR.
I left 2 more comments, please add a https://github.com/PostHog/posthog-android/blob/main/CHANGELOG.md#next entry under the Next section.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,9 +1,12 @@
 ## Next
 
+- allow anonymous uuid generation to be customizable ([#132](https://github.com/PostHog/posthog-android/pull/132))
+
 ## 3.2.2 - 2024-05-21
 
 - chore: register to sdk console ([#131](https://github.com/PostHog/posthog-android/pull/131)) 
 
+
```
Comment (marandaneto) [inline CHANGELOG.md:9]:
```suggestion
```

**Review by marandaneto — APPROVED** [review]

**Pull request merged**